src/libsyntax/codemap.rs

```diff
@@ -697,6 +697,22 @@ impl CodeMap {
         sp
     }
 
+    /// Given a `Span`, get a shorter one until `predicate` yields false.
+    pub fn span_take_while<P>(&self, sp: Span, predicate: P) -> Span
+        where P: for <'r> FnMut(&'r char) -> bool
+    {
+        if let Ok(snippet) = self.span_to_snippet(sp) {
+            let offset = snippet.chars()
+                .take_while(predicate)
+                .map(|c| c.len_utf8())
+                .sum::<usize>();
+
+            sp.with_hi(BytePos(sp.lo().0 + (offset as u32)))
+        } else {
+            sp
+        }
+    }
+
```

Comment:
Not sure if I should have done this here or inlined it in the function `suggest_remove_refs`. It seemed right to do it here... I am considering two options now:
- inline this function into the only place it's being used; or
- use this function in functions like `span_until_whitespace` which may use it.

@estebank what do you think?

Reply:
It's good to separate it, it makes the code above easier to follow.

The latter option sounds good to me if you don't mind doing it :)